homeassistant/components/nice_go/config_flow.py

```diff
@@ -66,3 +69,53 @@ async def async_step_user(
         return self.async_show_form(
             step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
         )
+
+    async def async_step_reauth(
+        self, entry_data: Mapping[str, Any]
+    ) -> ConfigFlowResult:
+        """Handle re-authentication."""
+        self.reauth_entry = self.hass.config_entries.async_get_entry(
+            self.context["entry_id"]
+        )
+
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Confirm re-authentication."""
+        errors = {}
+
+        if TYPE_CHECKING:
+            assert self.reauth_entry is not None
+
+        if user_input is not None:
+            new_input = {**self.reauth_entry.data, **user_input}
```

Comment:
Why do we overlay the old data? If it's for the suggested password I don't think it's a good idea to bring the old password back in the form

Reply:
As the schema is the same I agree and I also think it would be good to remove the pwd from the form. 

Should the user be allowed to change the email address?

Reply:
It's fairly simple on my end to allow for that, want me to do that? Otherwise I can just make a new schema and remove the email field

Reply:
Is it possible in the native app/configuration to change the email address?
If yes then keep it, if no then I don't think they should be able to do it here either.